File: src/test/java/com/ject/studytrip/mission/fixture/DailyMissionFixture.java
package com.ject.studytrip.mission.fixture;

import com.ject.studytrip.mission.domain.factory.DailyMissionFactory;
import com.ject.studytrip.mission.domain.model.DailyMission;
import com.ject.studytrip.mission.domain.model.Mission;
import com.ject.studytrip.trip.domain.model.DailyGoal;
import org.springframework.test.util.ReflectionTestUtils;

public class DailyMissionFixture {

    public static DailyMission createDailyMission(Mission mission, DailyGoal dailyGoal) {
        return DailyMissionFactory.create(mission, dailyGoal);
    }

    public static DailyMission createDailyMissionWithId(
            Long id, Mission mission, DailyGoal dailyGoal) {
        DailyMission dailyMission = DailyMissionFactory.create(mission, dailyGoal);
        ReflectionTestUtils.setField(dailyMission, "id", id);

        return dailyMission;
    }
}

File: src/test/kotlin/com/ject/studytrip/mission/fixture/DailyMissionFixture.kt
package com.ject.studytrip.mission.fixture

import com.ject.studytrip.mission.domain.factory.DailyMissionFactory
import com.ject.studytrip.mission.domain.model.DailyMission
import com.ject.studytrip.mission.domain.model.Mission
import com.ject.studytrip.trip.domain.model.DailyGoal
import org.springframework.test.util.ReflectionTestUtils

class DailyMissionFixture(
    private val mission: Mission,
    private val dailyGoal: DailyGoal,
) {
    fun create(): DailyMission = DailyMissionFactory.create(mission, dailyGoal)

    fun createWithId(id: Long): DailyMission =
        create().also {
            ReflectionTestUtils.setField(it, "id", id)
        }
}
